tests/integ/test_udf.py

```diff
@@ -1448,6 +1455,97 @@ def add_one_df_pandas_annotation(df: pandas.DataFrame) -> pandas.Series:
     )
 
 
+@pytest.mark.skipif(not is_pandas_and_numpy_available, reason="pandas is required")
+@pytest.mark.parametrize(
+    "_type, data, expected_type",
+    [
+        (IntegerType, [[4096]], "object"),  # TODO: should be int16
+        (IntegerType, [[1048576]], "object"),  # TODO: should be int32
+        (IntegerType, [[8589934592]], "object"),  # TODO: should be int64
+        (FloatType, [[1.0]], "float64"),
+        (StringType, [["1"]], "string"),
+        (BooleanType, [[True]], "boolean"),
+        (BinaryType, [[(1).to_bytes(1, byteorder="big")]], "object"),
+        (DateType, [[datetime.date(2021, 12, 20)]], "datetime64[ns]"),
+        (ArrayType, [[[1]]], "object"),
+        (TimeType, [[datetime.time(1, 1, 1)]], "timedelta64[ns]"),
+        (
+            TimestampType,
+            [[datetime.datetime(2016, 3, 13, 5, tzinfo=datetime.timezone.utc)]],
+            "datetime64[ns]",
+        ),
+        (GeographyType, [["POINT(30 10)"]], "object"),
```

Comment:
you missed MapType and VariantType.

We can have another test function for VariantType, because variant data itself has many date types (it's dynamically encoded/decoded).